debugger/gdb-sgx-plugin/debug_ext.py

```diff
@@ -0,0 +1,306 @@
+import gdb
+import math
+import tempfile
+from typing import List
+
+DEBUG = False
+
+INSTANCE_DEBUG_EXT = None
+
+class ThreadCloneBreakpoint(gdb.Breakpoint):
+    def __init__(self):
+        """
+        Set a breakpoint @ kernel/syscall.c::_SYS_myst_clone
+        Choosing this location because this is where the parent process/thread creates new process/thread
+        We can get the parent pid/tid from here
+        """
+        super(ThreadCloneBreakpoint, self).__init__('kernel/syscall.c:4094', internal=False)
+
+    def stop(self):
+        # Get Parent thread info
+        ppid = int(gdb.parse_and_eval('myst_thread_self()->process->ppid'))
```

Comment:
It is better to avoid function calls in `gdb.parse_and_eval`. Executing enclave functions from within gdb is not guaranteed to work.

Instead, evaluate the function in code, assign it to a volatile local variable and then read that variable from within the debugger. Even better would be to read a register instead of a named variable.

Reply:
We could make such modifications if we think the information from this breakpoint is important (i.e whos the parent thread/process)

Reply:
I've answered this, but it doesn't seem to show up correctly in github.

The key point is that gdb and lldb cannot call enclave functions like `myst_thread_self()` since it is not supported. Such function calls may or may-not work.

Reply:
This breakpoint is not actually in use, I will delete it altogether